include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1063 @@
+            );
+
+            /** all non-reduce-root processes are done now */
+            if( ! isPlaneReduceRoot )
+                return;
+
+            // gather to file writer
+            container::HostBuffer< float_64, DIM1 > globalSumMom2( subGrid.getGlobalDomain( ).size.y( ) );
+            container::HostBuffer< float_64, DIM1 > globalSumPos2( subGrid.getGlobalDomain( ).size.y( ) );
+            container::HostBuffer< float_64, DIM1 > globalSumMomPos( subGrid.getGlobalDomain( ).size.y( ) );
+            container::HostBuffer< float_64, DIM1 > globalCount_e( subGrid.getGlobalDomain( ).size.y( ) );
+
+            // gather y offsets, so we can store our gathered data in the right order
+            int gatherSize = -1;
+            int gatherRank = -1;
+            MPI_CHECK( MPI_Comm_rank( commGather, &gatherRank ) );
+            MPI_CHECK( MPI_Comm_size( commGather, &gatherSize ) );
+            std::vector< int > y_offsets( gatherSize );
+            std::vector< int > y_sizes( gatherSize );
+            std::vector< int > gatherRanks( gatherSize );
+            long int const y_off = subGrid.getLocalDomain( ).offset.y( );
+            int const y_siz = subGrid.getLocalDomain( ).size.y( );
+
+            MPI_CHECK( MPI_Gather(
+                &y_off,
+                1,
+                MPI_INT,
+                y_offsets.data( ),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ) );
+            MPI_CHECK( MPI_Gather(
+                &y_siz,
+                1,
+                MPI_INT,
+                y_sizes.data( ),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ) );
+            MPI_CHECK( MPI_Gather(
+                &gatherRank,
+                1,
+                MPI_INT,
+                gatherRanks.data( ),
+                1,
+                MPI_INT,
+                0,
+                commGather
+            ) );
+
+
+            std::vector< int > recvcounts( gatherSize, 1 );
+
+            MPI_CHECK( MPI_Gatherv(
+                gSumMom2->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain( ).size.y( ),
+                MPI_DOUBLE,
+                globalSumMom2.getDataPointer( ),
+                y_sizes.data( ),
+                y_offsets.data( ),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ) );
+            MPI_CHECK( MPI_Gatherv(
+                gSumPos2->getHostBuffer( ).getBasePointer( ),
+                subGrid.getLocalDomain( ).size.y( ),
+                MPI_DOUBLE,
+                globalSumPos2.getDataPointer( ),
+                y_sizes.data( ),
+                y_offsets.data( ),
+                MPI_DOUBLE,
+                0,
+                commGather
+            ) );
+            MPI_CHECK( MPI_Gatherv(
+                gSumMomPos->getHostBuffer( ).getBasePointer( ),
```

Comment:
**BUG BUG BUG**
This must be `reducedSumMomPos` and not the data from the gpu of the plain master. This is wrong for all buffer those reduce the plain data to the global y-buffer.

short: The global buffer contains **only** data from the plain master gpu.